pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -346,12 +359,23 @@ func (r *hibernationReconciler) setHibernatingCondition(cd *hivev1.ClusterDeploy
 			controllerutils.UpdateConditionIfReasonOrMessageChange,
 		)
 	}
+
+	if reason == hivev1.SyncSetsNotAppliedReason {
+		defer func() {
+			expiry := cd.Status.InstalledTimestamp.Time.Add(hibernateAfterSyncSetsNotApplied)
+			requeueAfter := time.Until(expiry)
+			logger.Infof("cluster will reconcile due to syncsets not applied in: %v", requeueAfter)
+			result.RequeueAfter = requeueAfter
+			result.Requeue = true
+		}()
```

Comment:
Looked like this might run after the if, but I looked it up and it appears it probably runs after the function we're in, so :+1: